matchms/utils.py

```diff
@@ -73,3 +73,25 @@ def _load_key_conversions(file: str, source: str, target: str) -> dict:
 def fingerprint_export_warning(spectrums: List[SpectrumType]):
     if any(x.get("fingerprint") is not None for x in spectrums):
         logger.warning("fingerprint found but will not be written to file.")
+
+
+def return_non_existing_file_name(file_name):
+    """Checks if a path already exists, otherwise creates a new filename with (1)"""
+    if not os.path.exists(file_name):
+        return file_name
+    print(f"The file name already exists: {file_name}")
+    file_name_base, file_extension = os.path.splitext(file_name)
+    i = 1
+    new_file_name = f"{file_name_base}({i}){file_extension}"
+    while os.path.exists(new_file_name):
+        i += 1
+        new_file_name = f"{file_name_base}({i}){file_extension}"
+    print(f"Instead the file will be stored in {new_file_name}")
+    return new_file_name
```

Comment:
I find this highly problematic - either a user passes a valid path or should generate termporary filenames based on a timestamp or so ... but

Reply:
The reason I added this, is that I sometimes rerun a pipeline and forget to remove the already existing file. It is really frustrating if you have a pipeline running overnight, finishing and not saving the file with the spectra, because it already existed. This always worked well for me as a quick fix.. But, using timestamps (or checking at the start) would of course work as well. 

Again my motivation was, since I find that it makes running more user friendly for me this probably is the case for other users as well and would therefore be good to add to matchms.

However, if this gives any issues for other pipelines, there is no need to add this to matchms of course and it is fine for me to remove it.

Reply:
I agree that this is not ideal but the pain Niek describes is probably even harder :)
Ideally, we still would like to avoid this I think and test before any extensive computation if file name issues are expected to arise.

If that is not easily possible I would suggest to keep such a solution for now, we make a clear issue that this should be avoided if possible as soon as we find better ways.
In that case, I would still change the function to make clearer (I hope) what it does. Maybe something like:
```python
def get_unique_filename(file_name: str):
    """
    Returns a unique filename, appending an index if the provided filename already exists.

    Parameters
    ------------
    file_name: Desired file name.

    Returns:
    - str: Unique file name.
    """
    if not os.path.exists(file_name):
        return file_name
    
    file_name_base, file_extension = os.path.splitext(file_name)
    i = 1

    while os.path.exists(f"{file_name_base}({i}){file_extension}"):
        i += 1
        
    return f"{file_name_base}({i}){file_extension}"
```

Reply:
What happens if this function is used could be defined at the location where it is in applied. 

```python
original_name = "sample.txt"
new_name = get_unique_filename(original_name)

if original_name != new_name:
    print(f"The file name '{original_name}' already exists.")
    print(f"Instead, the file will be stored as '{new_name}'.")
    # or do some logging etc...
```

Reply:
That is a nice cleanup indeed Florian. But I do agree that this is a bit too hacky, therefore I will just use this only locally if I have this issue when rerunning many times (mostly during testing). 
A better solution is to just raise an error, before the pipeline starts running. This was not very straightforward to implement, since the saving now happens outside of pipeline. But we can easily move this inside the Pipeline class and add a check if the file actually exists. I will remove these functions and implement the check and saving in pipeline instead.